implementations/rust/ockam/ockam_command/src/identity/show.rs

```diff
@@ -51,49 +51,94 @@ impl ShowCommand {
+                "Select one or more identities that you want to show".to_string(),
+                id_names,
+            );
+
+            if names.is_empty() {
+                opts.terminal
+                    .stdout()
+                    .plain("No identities selected")
+                    .write_line()?;
+                return Ok(());
+            }
+
+            if !opts
+                .terminal
+                .confirm_interactively(format!("Would you like to show these items : {:?}?", names))
+            {
+                opts.terminal
+                    .stdout()
+                    .plain("The action has not been confirmed")
+                    .write_line()?;
+                return Ok(());
             }
         } else {
-            let identifier_display = IdentifierDisplay(identifier);
-            (
-                identifier_display.to_string(),
-                to_string_pretty(&json!({"identifier": &identifier_display})),
-            )
-        };
-
-        opts.terminal
-            .stdout()
-            .plain(&plain)
-            .json(json.into_diagnostic()?)
-            .machine(&plain)
-            .write_line()?;
+            names.push("default".to_string());
+        }
+
+        for name in names {
+            let state = opts.state.identities.get(&name)?;
+            let identifier = state.config().identifier();
+            let (plain, json) = if cmd.full {
+                let change_history = opts
+                    .state
+                    .identities
+                    .identities_repository()
+                    .await?
+                    .get_identity(&identifier)
+                    .await
+                    .into_diagnostic()?;
+
+                if Some(EncodeFormat::Hex) == cmd.encoding {
+                    let encoded = hex::encode(change_history.export().into_diagnostic()?);
+                    let json = to_string_pretty(&json!({"encoded": &encoded}));
+                    (encoded, json)
+                } else {
+                    let identity: ShowIdentity = Identity::import_from_change_history(
+                        Some(&identifier),
+                        change_history,
+                        Vault::create_verifying_vault(),
+                    )
+                    .await
+                    .into_diagnostic()?
+                    .into();
+
+                    (identity.to_string(), to_string_pretty(&identity))
+                }
+            } else {
+                let identifier_display = IdentifierDisplay(identifier);
+                (
+                    identifier_display.to_string(),
+                    to_string_pretty(&json!({"identifier": &identifier_display})),
+                )
+            };
+
+            opts.terminal
+                .clone()
+                .stdout()
+                .plain(&plain)
+                .json(json.into_diagnostic()?)
+                .machine(&plain)
+                .write_line()?;
```

Comment:
@adrianbenavides . Looking at what ` identity  list` does. it gets its data from this
```
     let mut identities: Vec<IdentityListOutput> = Vec::new();
     let idts = opts.state.identities.list()?;
```
Is is acceptable to run this when a list output is required with show, filtering on the names that have been selected?

The `list` code relies on quite a lot more crates (async, mutex, some OckamColor code) that the `show` code . Would it make sense to factor out the listing code so that the same function/method can be used for by either `ockam identity show` or `ockam identity list`.?

Reply:
You can build the `IdentityListOutput` given the identity name and the `opts.state`:

```
IdentityListOutput {
    name,
    identifier: state.config().identifier(),
    is_default: opts.state.identities.is_default(&name)?,
}
```

Given the names array, you can iterate through it and return a new array of `IdentityListOutput`.

Reply:
Thank you.